### PR TITLE
diplomacy: provide connect access to edges without bundles

### PR DESCRIPTION
**Pull request by terpstra**
Forcing the bundles to exist early can mess up module ownership.

### CODE REVIEW

**Comment by terpstra** [comment]
If it works, feel free to merge it directly without waiting for me.

**Comment by jackkoenig** [comment]
This fixes my problem! Now that Travis is set up, need to get Travis to actually run it.

File: src/main/scala/uncore/tilelink2/Nodes.scala [hunk 1]
```diff
@@ -25,10 +25,10 @@ object TLImp extends NodeImp[TLClientPortParameters, TLManagerPortParameters, TL
...
-  override def connect(bindings: () => Seq[(TLEdgeIn, TLBundle, TLBundle)])(implicit p: Parameters, sourceInfo: SourceInfo): (Option[LazyModule], () => Unit) = {
-    val monitor = p(TLMonitorBuilder)(TLMonitorArgs(() => bindings().map(_._1), sourceInfo, p))
+  override def connect(edges: () => Seq[TLEdgeIn], bundles: () => Seq[(TLEdgeIn, TLBundle, TLBundle)])(implicit p: Parameters, sourceInfo: SourceInfo): (Option[LazyModule], () => Unit) = {
+    val monitor = p(TLMonitorBuilder)(TLMonitorArgs(edges, sourceInfo, p))
     (monitor, () => {
-      val eval = bindings ()
+      val eval = bundles ()
       monitor.foreach { m => (eval zip m.module.io.in) foreach { case ((_,i,o), m) => m := TLBundleSnoop(o,i) } }
       eval.foreach { case (_, bi, bo) =>
         bi <> bo
```
Comment (jackkoenig) [inline src/main/scala/uncore/tilelink2/Nodes.scala]: I see that you split edges and bundles, but why did you keep `TLEdgeIn` in the bundle Tuple? Unless I'm missing something, in both places where you iterate on bundles (as eval in lines 32 and 33), you ignore the `TLEdgeIn` field of the tuple.

Comment (terpstra) [inline src/main/scala/uncore/tilelink2/Nodes.scala]: I figured it could be useful for some cases. BTW, you did not review the PR.

**Comment by terpstra** [comment]
Included a few more commits while we wait for master to rebuild. @jackkoenig you still need to approve the PR to be able to merge it.

**Pull request merged**